src/models/msg.js

```diff
@@ -31,6 +31,7 @@ function Msg(attr) {
 	_.defaults(this, attr, {
 		from: "",
 		id: id++,
+		links: [],
 		previews: [],
```

Comment:
You can still take advantage of `previews` array`. By default you can set it to have `type` of either empty string or `loading`, and compare against that on the client. I think this will be slightly better than having a separate links array.